### PR TITLE
Site Migration: Update migration instructions screen

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #88588

## Proposed Changes
* Update the site migration instructions page to reflect the figma layout  5yhaC6umexdQLlLpCrCyEe-fi-2095_9011
* Update the instructions
* Automatically load the migration key without any user interaction
* Show a different message when we can’t get the migration link
* Add links directly to the related pages

More context: p1711029899474269-slack-C0Q664T29

## Testing Instructions
**SHORT VERSION:**
1. Enable the feature toggle
2. Access `/setup/site-migration/site-migration-instructions?from=[JN site]&siteSlug=[YOUR_WORDPRESS.COM site created by previous migrations]`
3. Jump to the steps 10, 11 and 12.

**LONG VERSION:**
1. Go to the `/start` URL using the calypso.live instructions
2. Follow the flow until see the "Goals" page
4. Open the browser Web developer tools and copy and paste the following code `sessionStorage.setItem('flags', "onboarding/new-migration-flow")`
5. Type Enter
6. Reload the goals page
7. Select "Import or migrate content"
8. "Where will you import from?" set `https://test-vanilla-wp.godaddy.test-wpcom-migrations.net/`
9. Select "Everything (requires a Creator Plan)" on "What do you want to migrate?"
10. When you arrive at the "Ready to migrate your site?" follow the instructions using the page links and copying the migration key. **Screenshot 1(a,b)**
11. After following the instructions, refresh the page to see a screen variation. **Screenshot 2(a,b)**
12. Check again the instructions for the new version, using the links.

NOTE: The screen variation with the migration key is only shown the first time the user sees the screen, after a reload only the second variation is shown. 

NOTE 2: If you want to reset the site state to see screen 1 again,  you just need to open the URL([TARGET_SITE]/_cli) and run the command `wp option delete wpcom_site_migration_migrate_guru_key_read` to delete the site option we are using to not show again the migration key.

### Screenshots
**1a. Desktop - Migration key available**
image

**1.b Mobile - Migration key available**
image

**2a. Desktop - Migration key not available**
image

**2b. Mobile - Migration key not available**
image

Interactions:

https://github.com/Automattic/wp-calypso/assets/38718/6082efbb-1069-4526-811b-aaba9e30b60c

### CODE REVIEW

**Comment by gabrielcaires** [comment]
@javierarce I added a video showing the animations and interactions but feel free to see the real screen.

**Comment by gabrielcaires** [comment]
@javierarce The show/hide eyes icon is different from the layout because there is a recommendation to use the wordpress/icons library 

https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library

image

**Comment by javierarce** [comment]
> @javierarce The show/hide eyes icon is different from the layout because there is a recommendation to use the wordpress/icons library

You are totally right, sorry about that, I'll update the icon in Figma.

Regarding the implementation of the design, it looks great, but if we made two little adjustments to the heights of the button and the input field, they'd look better. Could we make the height of the input field 44px and the height of the button 28px?

image

@michaelpick could you and Francisco review the copy of the instructions step (5yhaC6umexdQLlLpCrCyEe-fi-2095_9011)? I think I never showed them to you.

Here's the text:

> Ready to migrate your site?
> Follow these steps to get started.
>
> 1. Install and activate the Migrate Guru plugin on your existing site.
> 2. In the Migrate Guru plugin settings page, enter your email address and click Migrate.
> 3. When asked to select a destination host, pick “Automattic.”
> 4. Copy and paste the migration key below in the “Migrate Guru Migration Key” field and click Migrate.
> 
> And you are done! When the migration finishes, Migrate Guru will send you an email.

**Comment by gabrielcaires** [comment]
> Regarding the implementation of the design, it looks great, but if we made two little adjustments to the heights of the button and the input field, they'd look better. Could we make the height of the input field 44px and the height of the button 28px?
> 

image
image

@javierarce size updated! 

I also prevented the component from automatically selecting the key when it is hidden.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (valterlorran) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Can we set the name of the host to "WordPress.com"? They already updated it.

**Comment by javierarce** [comment]
> @javierarce size updated!
> I also prevented the component from automatically selecting the key when it is hidden.

Fantastic, much better, thanks!

**Comment by javierarce** [comment]
One last thing that I noted (but we can update this after we get the content review) is the use of italics. Since we are using quotes the italics are redundant.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Typo.
```suggestion
						'Install and activate the {{a}}Migrate Guru plugin{{/a}} on your existing site.',
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Question: are the label and button CTA translated in the Blog Vault/Migrate Guru UI?
We run the risk of translating these terms when they are not translated in the underlying screen.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Minor note (which is missing _all_ the context!): might it be clearer if we used "on ~the~ your source site"?

**Review by daledupreez — COMMENTED** [review]
I have some quick feedback from inspection.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Updated here  cc012deec311e73a0a9e059bd637364e965f9f60

**Comment by gabrielcaires** [comment]
> One last thing that I noted (but we can update this after we get the content review) is the use of italics. Since we are using quotes the italics are redundant.\

@javierarce Sure! I removed the quotes here 9f9478c4695797b0feef80530d50deebdbc9c125

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Typo fixed here https://github.com/Automattic/wp-calypso/pull/88813/commits/658ae3d6c4e12a09c2d425af37902f8c47574824

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Fixed here https://github.com/Automattic/wp-calypso/pull/88813/commits/658ae3d6c4e12a09c2d425af37902f8c47574824

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Good question!
I checked and saw they were not translating the CTAs, so I applied this change https://github.com/Automattic/wp-calypso/pull/88813/commits/d22bc4dfa381493da09fd3ac2b3f2df5b83a0202 to protect then to be translated.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (andres-blanco) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: What's the reasoning behind adding a `DoNotTranslateIt` component? 

maybe adding the markup in the input string directly and passing the `Migrate Guru Migration key` | `Migrate` strings as an [arg](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/types/index.d.ts#L34) is enough?

Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: @andres-blanco 
The idea is to express the intent the value should not be translated, especially considering we don't have any kind of automated test testing translations.

My concern is we can easily remove the args value just because the intent is not clear and we don't have a test suite testing translations.

Comment (andres-blanco) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: ok, then my only comment is that it would be good to remove the markup from the `DoNotTranslateIt` element and keep it in the input string. That way is more readable, and also extendable (imagine if you want to do something like `<strong><em>STRING</em></strong>`).
But it's just a nitpick, not a blocker. Go with your gut :)

Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx]: Good point; I tested, and we can still do it via `{{strong}} {{migrationKeyField /}} {{/ strong }}.` but I still agree with you because it blocks the translators to freely move the formatting position. 
So, I update here https://github.com/Automattic/wp-calypso/pull/88813/commits/17eefe3a9baba824878040c44fb9501043d416b1

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (daledupreez) [inline client/landing/stepper/hooks/use-site-migraiton-key.ts:1]: Nit: the filename has a typo. `-migraiton-` => `-migration-`

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (daledupreez) [inline client/landing/stepper/hooks/use-site-migraiton-key.ts:21]: Should we specify `staleTime` on the `useQuery()` call to allow the data to be kept in memory for a little while before we try to refetch? ([reference](https://tanstack.com/query/v4/docs/framework/react/reference/useQuery))

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (daledupreez) [inline client/landing/stepper/hooks/use-site-migraiton-key.ts:16]: Is the `!` intentional? If so, what does it do?!?

File: client/landing/stepper/hooks/test/use-site-migration-key.tsx
Comment (daledupreez) [inline client/landing/stepper/hooks/test/use-site-migration-key.tsx:23]: Nit: the argument expects the siteId argument to be numeric, so we should likely test it with a number (or allow a site slug to be specified in the types).

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/show-hide-input/style.scss
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/show-hide-input/style.scss:13]: Nit: I have a strong personal preference for avoiding `&` as selector/text duplicator in CSS selectors because it makes it much harder to find the selector when debugging in an IDE.

While that may not seem like a common use case, every now and then we hit issues with CSS selectivity in Calypso as a result of duplicate selector names that are defined in different CSS bundles, and knowing where and how the rules were defined is relatively important.

Also, I find I need to think extra-hard to make sure I know what `&` actually represents, especially within patch diffs where the wrapping selector may not be visible.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:87]: I don't have a super-strong opinion on this, but I do think it would still be relatively easy and clear if we handle this with args and a comment. (We may need the comment anyway.) e.g.
```suggestion
					{ translate(
						'Go to the {{a}}Migrate Guru page on your source site{{/a}}, enter your email address, and click {{strong}}%(migrate)s{{/strong}}.',
						{
							comment: 'Note that the migrate arg will always be "Migrate", as the Blog Vault UI is not translated',
							args: {
								migrate: 'Migrate',
							},
							components: {
								a: <a href={ getMigrateGuruPageURL( fromUrl ) } target="_blank" rel="noreferrer" />,
								strong: <strong />,
							},
						}
					) }
```

**Review by daledupreez — APPROVED** [review]
Overall, this is LGTM, though I do have some non-blocking feedback.

**Comment by javierarce** [comment]
> @javierarce Sure! I removed the quotes here https://github.com/Automattic/wp-calypso/commit/9f9478c4695797b0feef80530d50deebdbc9c125

Oh, I meant to remove the italics and leave the quotes. We don't italicize that type of text.

I've pinged Francisco to do a content review of the text. If possible, let's wait for it before we merge this PR.

**Comment by javierarce** [comment]
Ok, here are the corrections (1e1T4v-g_4Lu6Kv2uCRIRYGr7KAlqlv0hxcpTm34FWbA-gdoc):

image

```
Title: Ready to migrate your site?
Subtitle: Follow these steps

1. Install and activate the [Migrate Guru plugin] on your existing site.
2. Go to the [Migrate Guru page on your source site], enter your email address, and click **Migrate**.
3. When asked to select a destination host, pick ‘WordPress.com‘
4. Copy and paste the migration key below in the ‘Migrate Guru Migration Key‘ field and click **Migrate** once more.

And you are done! When the migration is ready, Migrate Guru will send you an email.
```

image

```
Title: Ready to migrate your site?
Subtitle: Follow these steps

1. Install and activate the [Migrate Guru plugin] on your existing site.
2. Go to the [Migrate Guru page on your source site], enter your email address, and click Migrate.
3. When asked to select a destination host, pick ‘WordPress.com‘
4. Head to the [Migrate Guru page on the new WordPress.com site] and copy the migration key. 
5. Paste the migration key on the ‘Migrate Guru Migration Key’ field of your existing site and click **Migrate** once more.

And you are done! When the migration is ready, Migrate Guru will send you an email.
```

**Comment by gabrielcaires** [comment]
@javierarce @daledupreez Due to some github bug I only could see your suggestions after I merged. 
I am going to open a new PR to fix your comment.

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (gabrielcaires) [reply client/landing/stepper/hooks/use-site-migraiton-key.ts:1]: Fixed https://github.com/Automattic/wp-calypso/pull/88952/commits/694f4fccc9098322d923d169e7b9f886fe9d15f3

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (gabrielcaires) [reply client/landing/stepper/hooks/use-site-migraiton-key.ts:16]: Yes, It is saying that typescript can assume the siteId will be available when the function is executed. 
The enabled flag is already ensuring it.

The line above ensures that react-query only triggers the queryFn when the siteId is available. 
``` 
enabled: !! siteId,
`` `

File: client/landing/stepper/hooks/test/use-site-migration-key.tsx
Comment (gabrielcaires) [reply client/landing/stepper/hooks/test/use-site-migration-key.tsx:23]: Fixed here https://github.com/Automattic/wp-calypso/pull/88952/commits/67c80a1df84213e1ab9cc8af355b9eba6d327ef9

We are with the typescript checking disabled on the test files on our pipeline😢 .

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/show-hide-input/style.scss
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/show-hide-input/style.scss:13]: Thanks for the context:
I removed https://github.com/Automattic/wp-calypso/pull/88952/commits/b24b270f9e140f52eb4ea1f4dc624f7a8f1e3b21 the usage of `&`

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx
Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx:87]: The feedback I received was that the translators sometimes don't read the comments. 

Regarding the args, I explained my concern here. 
https://github.com/Automattic/wp-calypso/pull/88813#discussion_r1539303167

File: client/landing/stepper/hooks/use-site-migraiton-key.ts
Comment (gabrielcaires) [reply client/landing/stepper/hooks/use-site-migraiton-key.ts:21]: I set 60 seconds as staleTime as we discussed on slack. 
https://github.com/Automattic/wp-calypso/pull/88952/commits/a8022e45bee59130156286d12da9f4d8ffe1a473

**Pull request merged**
